From 1d0b15cfc8e75e74eab6938263e29b44cdc6344a Mon Sep 17 00:00:00 2001
From: "Huang, Tao" <huangtao@rock-chips.com>
Date: Thu, 2 Mar 2017 14:32:33 +0800
Subject: [PATCH] arm64: dts: rockchip: move more common nodes to
 rk3368-android

Also fix io_domains and pmu_io_domains on rk3368-tb.dtsi

Change-Id: I90867a839079f67c68b8588304e06f7566749a3d
Signed-off-by: Huang, Tao <huangtao@rock-chips.com>
---
 .../boot/dts/rockchip/rk3368-android.dtsi     | 48 +++++++++++++
 arch/arm64/boot/dts/rockchip/rk3368-tb.dtsi   | 69 +++++--------------
 2 files changed, 66 insertions(+), 51 deletions(-)

diff --git a/arch/arm64/boot/dts/rockchip/rk3368-android.dtsi b/arch/arm64/boot/dts/rockchip/rk3368-android.dtsi
index d30a80ea2cbe..bb188b489054 100644
--- a/arch/arm64/boot/dts/rockchip/rk3368-android.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3368-android.dtsi
@@ -48,6 +48,54 @@
 		lcdc = &lcdc;
 	};
 
+	chosen {
+		bootargs = "earlycon=uart8250,mmio32,0xff690000 swiotlb=1";
+	};
+
+	fiq_debugger: fiq-debugger {
+		compatible = "rockchip,fiq-debugger";
+		rockchip,serial-id = <2>;
+		rockchip,signal-irq = <186>;
+		rockchip,wake-irq = <0>;
+		rockchip,irq-mode-enable = <1>;  /* If enable uart uses irq instead of fiq */
+		rockchip,baudrate = <115200>;  /* Only 115200 and 1500000 */
+		pinctrl-names = "default";
+		pinctrl-0 = <&uart2_xfer>;
+	};
+
+	reserved-memory {
+		#address-cells = <2>;
+		#size-cells = <2>;
+		ranges;
+
+		/* global autoconfigured region for contiguous allocations */
+		linux,cma {
+			compatible = "shared-dma-pool";
+			reusable;
+			size = <0x0 0x8000000>;
+			linux,cma-default;
+		};
+
+		/* reg = <0x0 0x0 0x0 0x0> will be updated by uboot */
+		rockchip_logo: rockchip-logo@00000000 {
+			compatible = "rockchip,fb-logo";
+			reg = <0x0 0x0 0x0 0x0>;
+		};
+	};
+
+	ion {
+		compatible = "rockchip,ion";
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		cma-heap {
+			reg = <0x00000000 0x02000000>;
+		};
+
+		system-heap {
+		};
+	};
+
 	isp: isp@ff910000 {
 		compatible = "rockchip,rk3368-isp", "rockchip,isp";
 		reg = <0x0 0xff910000 0x0 0x10000>;
diff --git a/arch/arm64/boot/dts/rockchip/rk3368-tb.dtsi b/arch/arm64/boot/dts/rockchip/rk3368-tb.dtsi
index c35a66fb5c62..f925872211d9 100644
--- a/arch/arm64/boot/dts/rockchip/rk3368-tb.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3368-tb.dtsi
@@ -47,37 +47,6 @@
 / {
 	compatible = "rockchip,tb", "rockchip,rk3368";
 
-	chosen {
-		bootargs = "console=uart,mmio32,0xff690000";
-	};
-
-	reserved-memory {
-		#address-cells = <2>;
-		#size-cells = <2>;
-		ranges;
-
-		/* global autoconfigured region for contiguous allocations */
-		linux,cma {
-			compatible = "shared-dma-pool";
-			reusable;
-			size = <0x0 0x8000000>;
-			linux,cma-default;
-		};
-	};
-
-	ion {
-		compatible = "rockchip,ion";
-		#address-cells = <1>;
-		#size-cells = <0>;
-
-		cma-heap {
-			reg = <0x00000000 0x02000000>;
-		};
-
-		system-heap {
-		};
-	};
-
 	sound {
 		compatible = "simple-audio-card";
 		simple-audio-card,format = "i2s";
@@ -197,26 +166,6 @@
 		regulator-min-microvolt = <3800000>;
 		regulator-max-microvolt = <3800000>;
 	};
-
-	io-domains {
-		compatible = "rockchip,rk3368-io-voltage-domain";
-		rockchip,grf = <&grf>;
-
-		dvp-supply = <&vcc_18>;
-		audio-supply = <&vcc_io>;
-		gpio30-supply = <&vcc_io>;
-		gpio1830-supply = <&vcc_io>;
-		sdcard-supply = <&vccio_sd>;
-		wifi-supply = <&vccio_wl>;
-	};
-
-	pmu-io-domains {
-		compatible = "rockchip,rk3368-pmu-io-voltage-domain";
-		rockchip,grf = <&pmugrf>;
-
-		pmu-supply = <&vcc_io>;
-		vop-supply = <&vcc_io>;
-	};
 };
 
 &emmc {
@@ -618,6 +567,24 @@
 	#sound-dai-cells = <0>;
 };
 
+&io_domains {
+	status = "okay";
+
+	dvp-supply = <&vcc_18>;
+	audio-supply = <&vcc_io>;
+	gpio30-supply = <&vcc_io>;
+	gpio1830-supply = <&vcc_io>;
+	sdcard-supply = <&vccio_sd>;
+	wifi-supply = <&vccio_wl>;
+};
+
+&pmu_io_domains {
+	status = "okay";
+
+	pmu-supply = <&vcc_io>;
+	vop-supply = <&vcc_io>;
+};
+
 &pwm0 {
 	status = "okay";
 };
-- 
2.35.3

